server/chat-plugins/room-events.ts

```diff
@@ -30,19 +53,25 @@ function formatEvent(event: RoomEvent, showAliases?: boolean) {
 function getAllAliases(room: Room) {
 	if (!room.settings.events) return [];
 	const aliases: string[] = [];
-	for (const event of Object.values(room.settings.events)) {
-		if (event.aliases) aliases.push(...event.aliases);
+	for (const aliasID of Object.keys(room.settings.events).filter(id => 'eventID' in room.settings.events![id])) {
+		aliases.push(aliasID);
 	}
```

Comment:
tbh this is a weird mix of functional and regular code, and it makes more sense to pick one or the other.

```suggestion
	for (const aliasID in room.settings.events) {
		if ('eventID' in room.settings.events[aliasID]) aliases.push(aliasID);
	}
```

Reply:
If you instead wanted to make it purely functional, it would look like this:

```ts
return Object.entries(room.settings.events).filter(
  ([id, event]) => 'eventID' in event
).map(
  ([id]) => id
);
```

I don't have a strong opinion of which style to prefer; it's up to you.

Reply:
(Both styles support filtering by event ID roughly equally well.)